src/scheduler/worker.rs

```diff
@@ -394,16 +233,50 @@ impl<VM: VMBinding> GCWorker<VM> {
             probe!(mmtk, work, typename.as_ptr(), typename.len());
             work.do_work_with_stat(self, mmtk);
         }
+        debug!(
+            "Worker exiting. ordinal: {}, {}",
+            self.ordinal,
+            crate::util::rust_util::debug_process_thread_id(),
+        );
+        probe!(mmtk, gcworker_exit);
     }
 }
 
-/// A worker group to manage all the GC workers (except the coordinator worker).
+enum WorkerCreationState<VM: VMBinding> {
```

Comment:
Yes.  Since the enum variant names already indicate the current state, it is much clearer to simply remove the adjectives from the variables.  I'll also use the word "surrender" consistently.